### PR TITLE
#864 - check the SS and MS property values to be in options

### PR DESCRIPTION
**Pull request by raduiordache**
Fixes #<issue>

## Checklist:
- [x] My code follows the pykechain style
    - [x] I added type hinting to all functions
    - [x] I added documentation for all public functions
    - [x] I locally used `tox` to check for dists and docs errors
    - [x] My code passes the `pep8` and `flake8` linting checks and no warnings
    - [x] I removed unused imports, using `Code > Optimize Imports` in PyCharm
    - [x] I used [`black`](https://github.com/psf/black) to format the new code
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] I committed fresh test cassettes
    - [x] I have proper test coverage (don't decline the coverage of the test)
- [x] I asked another teammate to review the code
- [x] I update the Changelog.md with the appropriate changes.

### CODE REVIEW

File: pykechain/extra_utils.py [hunk 1]
```diff
@@ -273,25 +273,28 @@ def update_part_with_properties(part_instance, moved_instance, name=None):
...
+                moved_prop_instance.clear()
         # For a reference value property, add the id's of the part referenced {property.id: [part1.id, part2.id, ...]},
         # if there is part referenced at all.
         elif prop_instance.type == PropertyType.REFERENCES_VALUE:
             if prop_instance.value:
-                moved_prop_instance = get_mapping_dictionary()[prop_instance.id]
                 properties_id_dict[moved_prop_instance.id] = [ref_part.id for ref_part in prop_instance.value]
+        elif prop_instance.type in (PropertyType.SINGLE_SELECT_VALUE, PropertyType.MULTI_SELECT_VALUE):
+            if prop_instance.model().options:
+                if prop_instance.value in prop_instance.model().options:
```
Comment (JelleBoersma) [inline pykechain/extra_utils.py:295]: Does this work for multi-select values if `value` is a list?

**Review by JelleBoersma — CHANGES_REQUESTED** [review]

File: pykechain/extra_utils.py (hunk 1 above)
Comment (raduiordache) [reply pykechain/extra_utils.py:295]: Well spotted

File: pykechain/extra_utils.py [hunk 2]
```diff
@@ -290,10 +290,13 @@ def update_part_with_properties(part_instance, moved_instance, name=None):
         elif prop_instance.type == PropertyType.REFERENCES_VALUE:
             if prop_instance.value:
                 properties_id_dict[moved_prop_instance.id] = [ref_part.id for ref_part in prop_instance.value]
-        elif prop_instance.type in (PropertyType.SINGLE_SELECT_VALUE, PropertyType.MULTI_SELECT_VALUE):
+        elif prop_instance.type == PropertyType.SINGLE_SELECT_VALUE:
             if prop_instance.model().options:
                 if prop_instance.value in prop_instance.model().options:
                     properties_id_dict[moved_prop_instance.id] = prop_instance.value
+        elif prop_instance.type == PropertyType.MULTI_SELECT_VALUE:
+            if all(value in prop_instance.model().options for value in prop_instance.value):
```
Comment (JelleBoersma) [inline pykechain/extra_utils.py:298]: This will not work if value is `None`

**Review by JelleBoersma — CHANGES_REQUESTED** [review]

File: pykechain/extra_utils.py (hunk 2 above)
Comment (raduiordache) [reply pykechain/extra_utils.py:298]: I am retracing the same mistakes you did, ain't I?

**Review by JelleBoersma — APPROVED** [review]
Other than merging in master first, this is fine!

**Review by jberends — APPROVED** [review]

**Pull request merged**